src/CircleEditor/CircleEditorDocument.ts

```diff
@@ -191,4 +195,333 @@ export class CircleEditorDocument extends Disposable implements vscode.CustomDoc
+          Balloon.error("'boolean' type must be 'true' or 'false'.", false);
+          return;
+        }
+      }
+      else if (valType === "int") {
+        if (this.guessExactType(val) === 'float') {
+          Balloon.error("'int' type doesn't include decimal point.", false);
+          return;
+        }
+        fbb.addInt(Number(val));
+      }
+      else {
+        fbb.add(String(val));
+      }
+    }
+    fbb.end();
+    const res = fbb.finish();
+    const buf = Buffer.alloc(res.byteLength);
+    const view = new Uint8Array(res);
+    for (let i = 0; i < buf.length; ++i) {
+      buf[i] = view[i];
+    }
+    const data = Array.from(buf);
+    let responseData = {
+      command: 'responseEncodingData',
+      data: data
+    };
+    this._onDidChangeContent.fire(responseData);
+  }
+
+  /**
+   * Decode customOptions using flexbuffers when we click custom operator's node.
+   * Send value's type of custom operator. 
+   */
+  public sendCustomType(message: any) {
+    const msgData: any = message.data;
+    const subgraphIdx: number = msgData._subgraphIdx;
+    const operatorIdx: number = msgData._nodeIdx;
+    const target = this._model.subgraphs[subgraphIdx].operators[operatorIdx].customOptions;
+    const buffer = Buffer.from(target);
+    const ab = new ArrayBuffer(buffer.length);
+    const view = new Uint8Array(ab);
+    for (let i = 0; i < buffer.length; ++i) {
+      view[i] = buffer[i];
+    }
+    const customObj: any = flexbuffers.toObject(ab);
+    let resData: any = new Object;
+    resData._subgraphIdx = subgraphIdx;
+    resData._nodeIdx = operatorIdx;
+    resData._type = new Object;
+    for (const key in customObj) {
+      let customObjDataType: any = typeof (customObj[key]);
+      if (customObjDataType === 'number') {
+        customObjDataType = this.guessExactType(customObj[key]);
+      }
+      resData._type[key] = customObjDataType;
+    }
+    let responseData = {
+      command: 'customType',
+      data: resData
+    };
+
+    this._onDidChangeContent.fire(responseData);
+    return;
+  }
+
+  /**
+   * This function edit Tensor's name and shape, data's type of buffer, buffer's data.
+   */
+  private editTensor(data: any) {
+    let name;
+    let subgraphIdx: number = 0;
+    let argname: string;
+    let tensorIdx: number;
+    let isVariable: boolean = false;
+    let tensorType;
+    let tensorShape;
+    let bufferData: any = null;
+    name = data?._name;
+    subgraphIdx = Number(data._subgraphIdx);
+    if (name === undefined || name === undefined) {
+      Balloon.error("input data is undefined", false);
+      return;
+    }
+    const argument = data._arguments;
+    argname = argument._name;
+    tensorIdx = Number(argument._location);
+    const isChanged: boolean = argument._isChanged;
+    tensorType = argument._type._dataType;
+    tensorShape = argument._type._shape._dimensions;
+    if (name === undefined || name === undefined || name === undefined || name === undefined) {
```

Comment:
.....?

Reply:
I'm sorry. It's my mistake... I'll edit right now!!